drivers/bluetooth/hci/h4.c

```diff
@@ -1,45 +1,39 @@
 /* h4.c - H:4 UART based Bluetooth driver */
 
 /*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
  * Copyright (c) 2015-2016 Intel Corporation
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <errno.h>
-#include <stddef.h>
-
-#include <zephyr.h>
-#include <arch/cpu.h>
-
-#include <init.h>
-#include <drivers/uart.h>
-#include <sys/util.h>
-#include <sys/byteorder.h>
-#include <string.h>
-
+#include <drivers/bluetooth/h4_uart.h>
 #include <bluetooth/bluetooth.h>
 #include <bluetooth/hci.h>
 #include <drivers/bluetooth/hci_driver.h>
+#include <sys/byteorder.h>
 
 #define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_DRIVER)
 #define LOG_MODULE_NAME bt_driver
 #include "common/log.h"
 
 #include "../util.h"
 
-#define H4_NONE 0x00
-#define H4_CMD  0x01
-#define H4_ACL  0x02
-#define H4_SCO  0x03
-#define H4_EVT  0x04
+static struct h4_uart transport;
 
 static K_THREAD_STACK_DEFINE(rx_thread_stack, CONFIG_BT_RX_STACK_SIZE);
-static struct k_thread rx_thread_data;
+
+static __aligned(sizeof(uint32_t)) uint8_t
```

Comment:
yes, for async uart these buffers are required because uart use them directly. I'll check if in case of interrupt driven these can be fixed to some small number (will measure cpu load).